fs/fs.go

```diff
@@ -29,7 +29,7 @@ func NewDatastore(path string) (ds.Datastore, error) {
 
 // KeyFilename returns the filename associated with `key`
 func (d *Datastore) KeyFilename(key ds.Key) string {
-	return filepath.Join(d.path, key.String(), ObjectKeySuffix)
```

Comment:
`ds.Put("/foo/.object", "blah")` -- what now? Can't mkdir `foo/.object` as it's already a file.

You could give every directory level an extension too, but that gets ugly, and stops you from doing that `os.MkdirAll`.

The reason I thought this to be unintended is mostly that it wastes at least two inodes per object, and causes way more syscalls. It may be fine for what it is, but high performance it ain't.